apps/aggregator/src/snapshots.service.ts

```diff
@@ -32,23 +35,38 @@ export class SnapshotsService {
     const liquidStakingUsers: Record<string, BigNumber> = {};
 
     const network = this.apiConfigService.getNetwork();
-    for (const provider of snapshotsProviders) {
-      this.logger.log(`Started processing liquid staking for provider '${provider}'`);
+    for (const providerName of snapshotsProviders) {
+      const providerUsers: Record<string, BigNumber> = {};
+      const stakedValueSum: BigNumber = new BigNumber(0);
+      const providerInstance = await loadProvider(this.baseProvider, network, providerName);
+      if (!providerInstance) {
+        this.logger.error(`Cannot load scenario with name ${providerName}`);
+      }
+      this.logger.log(`Started processing liquid staking for provider '${providerName}'`);
       try {
-        const { stakedValueSum } = await this.fetchDataForProject(provider, network, liquidStakingUsers);
-        this.logger.log(`Total liquid staking for provider: ${provider}: ${stakedValueSum.dividedBy(1e18)} EGLD`);
+        const { stakedValueSum } = await this.fetchDataForProject(providerInstance, providerName, providerUsers);
+        this.logger.log(`Total liquid staking for provider: ${providerName}: ${stakedValueSum.dividedBy(1e18)} EGLD`);
       } catch (e) {
-        await this.alertsService.sendIndexerError(`Error while indexing data for ${provider}: ${e}`);
-        this.logger.error(`Error while indexing data for ${provider}: ${e}`);
+        await this.alertsService.sendIndexerError(`Error while indexing data for ${providerName}: ${e}`);
+        this.logger.error(`Error while indexing data for ${providerName}: ${e}`);
+        continue;
+      }
+
+
+      if (await this.isSumOfStakedLessOrEgualToContractStake(providerName, stakedValueSum, await providerInstance.getStakingContracts())) {
+        this.addProviderRecordsToSum(providerUsers, liquidStakingUsers);
+      } else {
+        this.logger.error(`Liquid Staking provider sum of users stake is larger than the staking contracts stake. Provider=${providerName}, Sum=${stakedValueSum.toString(10)}`);
       }
     }
 
+    this.jsonExporter.exportJson(liquidStakingUsers);
     await this.addRecordsToIndexer(liquidStakingUsers);
```

Comment:
also, maybe it would be nice to send a slack alert in case something goes wrong

Reply:
done